src/transformers/models/patchtst/modeling_patchtst.py

```diff
@@ -0,0 +1,1958 @@
+
+        return x_mask, mask
+
+
+class ChannelAttentionTSTEncoder(nn.Module):
+    def __init__(self, config: PatchTSTConfig):
+        super().__init__()
+
+        self.layers = nn.ModuleList([ChannelAttentionTSTEncoderLayer(config) for i in range(config.encoder_layers)])
+
+    def forward(self, src: torch.Tensor, output_hidden_states: Optional[bool] = None):
+        """
+        src: tensor [bs x nvars x sequence_length x d_model] Return:
+            Tensor [bs x nvars x sequence_length x d_model]
+        """
+        all_hidden_states = []
+
+        for mod in self.layers:
+            src = mod(src)
+            if output_hidden_states:
+                all_hidden_states.append(src)
+        if output_hidden_states is None:
+            return src, None
+        return src, all_hidden_states
+
+
+class ChannelAttentionTSTEncoderLayer(nn.Module):
+    def __init__(self, config: PatchTSTConfig):
+        super().__init__()
+
+        self.channel_attention = config.channel_attention
+        # Multi-Head attention
+        # self.self_attn = PatchTSTAttention(config)
+
+        self.self_attn = PatchTSTAttention(
+            embed_dim=config.d_model,
+            num_heads=config.encoder_attention_heads,
+            dropout=config.attention_dropout,
+        )
+
+        # Add & Norm of the sublayer 1
+        self.dropout_path1 = nn.Dropout(config.dropout_path) if config.dropout_path > 0 else nn.Identity()
+        if "batch" in config.norm.lower():
+            self.norm_sublayer1 = nn.Sequential(Transpose(1, 2), nn.BatchNorm1d(config.d_model), Transpose(1, 2))
+        else:
+            self.norm_sublayer1 = nn.LayerNorm(config.d_model)
+
+        # Add & Norm of the sublayer 2
+        if self.channel_attention:
+            self.dropout_path2 = nn.Dropout(config.dropout_path) if config.dropout_path > 0 else nn.Identity()
+            if "batch" in config.norm.lower():
+                self.norm_sublayer2 = nn.Sequential(Transpose(1, 2), nn.BatchNorm1d(config.d_model), Transpose(1, 2))
+            else:
+                self.norm_sublayer2 = nn.LayerNorm(config.d_model)
+
+        # Position-wise Feed-Forward
+        self.ff = nn.Sequential(
+            nn.Linear(config.d_model, config.encoder_ffn_dim, bias=config.bias),
+            ACT2CLS[config.activation_function](),
+            nn.Dropout(config.ff_dropout) if config.ff_dropout > 0 else nn.Identity(),
+            nn.Linear(config.encoder_ffn_dim, config.d_model, bias=config.bias),
+        )
+
+        # Add & Norm of sublayer 3
+        self.dropout_path3 = nn.Dropout(config.dropout_path) if config.dropout_path > 0 else nn.Identity()
+        if "batch" in config.norm.lower():
+            self.norm_sublayer3 = nn.Sequential(Transpose(1, 2), nn.BatchNorm1d(config.d_model), Transpose(1, 2))
+        else:
+            self.norm_sublayer3 = nn.LayerNorm(config.d_model)
+
+        self.pre_norm = config.pre_norm
+
+    def forward(self, src: torch.Tensor):
```

Comment:
for consistency with other models in the library, let's call src `hidden_state`

Reply:
Fixed